cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js

```diff
@@ -10,9 +10,39 @@ import {
   VariableGridSchema
 } from './styled';
 import getGridTemplate from './helpers';
+import { getFilters } from '../Toolbar/FilterPicker/helpers';
 
-export default function LogList({ logs, template }) {
+export default function LogList({ widgetLocalStorage, logs, template }) {
   const theme = useTheme();
+  const filters = getFilters(widgetLocalStorage);
+
+  const filterByRegExp = (log, filters) => {
```

Comment:
Filtrowanie miało się odbywać na backendzie. Filtry trzymamy na froncie (to jest dobrze), ale po zatwierdzeniu filtra musi iść call do backendu, żeby ten od tej pory wysyłał do mnie tylko filtrowane logi (ale składowanie bez zmian - do tego będzie kwarantanna). Dodatkowo martwi mnie nieoptymalność tej funkcji. Jak będziemy to robić na backendzie, to zadbajmy proszę o to, bo tutaj widzę sporo logiki i niepotrzebnych pętli, a funkcja jest aplikowana dla każdego loga (więcej logów = więcej czasu na przetworzenie)

Reply:
UPDATE 25.11: Wszystkie filtrowania robimy NA FRONCIE